src/helpers/SuggestionRegistry.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type Suggestion = {
+	value: string;
+	description?: string;
+	group?: string;
+};
+
+export type SuggestionContext = {
+	propertyName: string;
+	inputValue: string | number;
+	cursorPosition?: number | null;
+};
+
+export type SuggestionProviderFunction = (word: string, context: SuggestionContext) => Suggestion[] | Promise<Suggestion[]>;
```

Comment:
Shouldn't these types declared on kaoto side? Otherwise, Kaoto won't be able to call them without introducing a cyclic dependency or redeclaring the types (not even sure it would work in this case)

Reply:
(and from the VS Code Kaoto side, I think that we can consider that the form library is part of Kaoto)